tests/virtualization/universal/storage.pm

```diff
@@ -17,72 +18,40 @@ use utils;
 
     record_info "Pool define";
-    assert_script_run "virsh pool-define-as testing dir - - - - '/pool_testing'";
-
-    record_info "Pool list";
-    assert_script_run "virsh pool-list --all | grep testing";
-
-    record_info "Pool build";
-    assert_script_run "virsh pool-build testing";
-
-    record_info "Pool start";
-    assert_script_run "virsh pool-start testing";
-
-    record_info "Pool autostart";
-    assert_script_run "virsh pool-autostart testing";
-
-    record_info "Pool info";
-    assert_script_run "virsh pool-info testing";
-
-    record_info "Create";
-    assert_script_run("virsh vol-create-as testing $_-storage 100M", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run "virsh pool-define-as $dir_pool_name dir - - - - '/pool_testing'";
 
-    record_info "Listing";
-    assert_script_run("ls /pool_testing/", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh vol-list testing | grep $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+    ## Basic Dir Storage Management
+    my $dir_vol_size = '100M';
+    my $dir_vol_resize = '200M';
+    virt_storage_management($dir_pool_name, size => $dir_vol_size, dir => 1, resize => $dir_vol_resize);
 
-    record_info "Info";
-    assert_script_run("virsh vol-info --pool testing $_-storage", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Dump XML";
-    assert_script_run("virsh vol-dumpxml --pool testing $_-storage", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Resize";
-    assert_script_run("virsh vol-resize --pool testing $_-storage 200M", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Attached";
-    my $target = (is_xen_host) ? "xvdx" : "vdx";
-    assert_script_run("virsh attach-disk --domain $_ --source `virsh vol-path --pool testing $_-storage` --target $target", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh detach-disk $_ $target", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Clone";
-    assert_script_run("virsh vol-clone --pool testing $_-storage $_-clone", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh vol-info --pool testing $_-clone", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Remove";
-    assert_script_run("virsh vol-delete --pool testing $_-clone", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh vol-delete --pool testing $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+    ## Cleanup
+    # Destroy the Dir storage pool
+    destroy_virt_storage_pool($dir_pool_name);
+}
 
-    record_info "Pool destroy";
-    assert_script_run "virsh pool-destroy testing";
-    assert_script_run "virsh pool-delete testing";
-    assert_script_run "virsh pool-undefine testing";
+# Prepare Virtualization Dir Storage Pool source
+sub prepare_dir_storage_pool_source {
```

Comment:
@guoxuguang  I just realize that you also made this for directory type storage. 
The reusable storage functions are put in virtual_storage_utils, rather than a new baseclass.This overall change is different from my proposal that we define the test procedures( prepare storage source, define storage pool, pool and volume managementm clean up storage vol/pool/source) into a new base class's func `run_test` for storage, then overwrite the relevant steps( prepare storage source, define storage pool, pool and volume managementm clean up storage vol/pool/source)  for each inherited class in storage.pm, libvirt_extend_storage_lvm.pm, and potentially the other two to-be-implemented storage types in future. 

But the current way that you do is a workable way too, although maybe more obvious code redundancy for each storage class in test procedure definition when we have four types storage.

I am fine to have it merged just this way for this PR. We can go back to this class refinement once you implement next new type of storage. Thanks for the efforts!

Besides, since storage.pm is changed which MU tests use. It will be better to be merged after reviewed by @tonyyuan1 and  @tbaev .

Reply:
@alice-suse  great! thanks so much for your suggestion and comments. 
yes, we can go back to this class refinement once implement next new type of storage (iscsi-target and multipath).
Hope that continue to follow the relevant steps( prepare storage source, define storage pool, pool and volume management, clean up storage vol/pool/source)